### ISSUE TITLE
Setup CI - Github actions

### ISSUE THREAD

**Issue by tristanMatthias** [code-mentoring/learn#7]
**Is your feature request related to a problem? Please describe.**
Currently there is no CI to check PRs and new commits.

**Describe the solution you'd like**
Using Github actions, we should have an automated check that tests the API and front end, linting, seeding, etc for any new PR or commit

**Describe alternatives you've considered**
Travis could also work, but github actions seems to be the best solution considering our tasks are all here

**Comment by shane215** [comment]
I'll look into this.
I'm familiar with the CI process, but I'm new to being hands on setting it up, so help appreciated on this.

**Comment by tristanMatthias** [comment]
No worries @shane215 . We're going to need this sooner than we thought actually. We need a way of publishing the `@code-mentoring/ui` package so it can be used in https://github.com/code-mentoring/www

**Comment by shane215** [comment]
@tristanMatthias 
I'm reading up on GitHub Actions and workflows now.
I imagine that we need on push workflows to build and run tests for each feature branch; at a minimum.

**Comment by tristanMatthias** [comment]
Yes at a minimum. Setup the basic structure etc. Then adding an extra command should be trivial (IE: adding linting check)

**Comment by shane215** [comment]
I'm imagining the basic structure as just build and run unit tests, with a command something like

`action "npm test" {
  uses = "actions/npm@master"
  args = ["test"]
  needs = ["npm install"]
}`

I'm still very ignorant of this, looking into it.

**Comment by shane215** [comment]
I found this basic setup documentation:
https://help.github.com/en/actions/building-and-testing-code-with-continuous-integration

**Comment by shane215** [comment]
I need some help with this. Specifically, I'm not sure how to move forward from this point, and I think that we use yarn instead of npm commands.

I followed [this documentation](https://help.github.com/en/actions/building-and-testing-code-with-continuous-integration/setting-up-continuous-integration-using-github-actions) and so far I have "ci.yml" (see below) which will be placed in "learn/.github/workflows/" in the repo.

-----
# This is a basic workflow to help you get started with Actions

name: CI

# Controls when the action will run. Triggers the workflow on push or pull request
# events but only for the master branch
on:
  push:
    branches: 
    - master
    - release/*
  pull_request:
    branches: 
    - master

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "build"
  build:
    # The type of runner that the job will run on
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        node_version: [8, 10, 12]
        os: [ubuntu-latest, windows-latest, macOS-latest]

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
    - uses: actions/checkout@master

    # Runs a set of commands using the runners shell
    - name: build and test
    - run: |
        npm install
        npm run build --if-present
        npm test
-----

**Comment by shaneSheridan** [comment]
BLOCKED Reason: Test script failing until we have an in memory test database.

**Comment by shaneSheridan** [comment]
UNBLOCKED Reason: Disabled Test command for now.
